### PR TITLE
#31: Introducing Basic API Integration guide and Starter Kit Collection.

### PR DESCRIPTION
**Pull request by MS-ResilienX**

### CODE REVIEW

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 1]
```diff
@@ -0,0 +1,2038 @@
...
+								"url": {
+									"raw": "{{baseUrl}}{{Monitoring}}/component",
+									"host": [
+										"{{baseUrl}}{{Monitoring}}"
+									],
+									"path": [
+										"component"
+									]
+								},
+								"description": "This information allows FRAIHMWORK to track basic information about a component. Any extra or custom fields are expected to always be present even if null values are provided at first. Using this endpoint means that the component needs to be registered (under the hood), so a UUID will be created and assigned in the response"
```
[the remark this answers is not in the document]
Comment (GD-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:142]: I think this documentation field would be a good spot to indicate that the UUID field is optional, and that leaving it blank will result in a UUID being generated for you

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 2]
```diff
@@ -0,0 +1,2038 @@
+{
```
[the remark this answers is not in the document]
Comment (GD-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1]: Might also be worth mentioning that the intent of this starter kit would be to have folks fill out the collection variables, peruse the different calls, and see how data is formed, not to just run the whole thing at once and feel good about themselves :P

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 3]
```diff
@@ -0,0 +1,2038 @@
...
+											],
+											"variable": [
+												{
+													"key": "uuid",
+													"value": "{{UUIDComponentResponse}}",
+													"description": "(Required) Uuid of the component properties being updated"
+												}
+											]
+										},
+										"description": "This information allows FRAIHMWORK to track basic information about a component. Any extra or custom fields are expected to always be present even if null values are provided at first."
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:297]: This description is not accurate - any fields left out / set to null in a PUT are telling FRAIHMWORK that 'there is no change to this field'. A PUT command to this endpoint with all of the optional fields left out is exactly equivalent to a refresh command.

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 4]
```diff
@@ -0,0 +1,2038 @@
...
+								"url": {
+									"raw": "{{baseUrl}}{{Monitoring}}/fault",
+									"host": [
+										"{{baseUrl}}{{Monitoring}}"
+									],
+									"path": [
+										"fault"
+									]
+								},
+								"description": "Allows a component to report one or more faults currently affecting the component. Using this endpoint means that the client is not managing their own fault IDs, and the fault will require a new tracking UUID (that will be created and assigned in the response)."
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:482]: I think the documentation here is referencing the original API spec, and needs updating

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 5]
```diff
@@ -0,0 +1,2038 @@
...
+											],
+											"variable": [
+												{
+													"key": "uuid",
+													"value": "{{UUIDFaultResponse}}",
+													"description": "(Required) Uuid of the fault being updated"
+												}
+											]
+										},
+										"description": "Allows a component to report one or more faults currently affecting the component"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:741]: I think the documentation here is referencing the original API spec, and needs updating

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 6]
```diff
@@ -0,0 +1,2038 @@
...
+								"url": {
+									"raw": "{{baseUrl}}{{Mitigation}}/mitigation",
+									"host": [
+										"{{baseUrl}}{{Mitigation}}"
+									],
+									"path": [
+										"mitigation"
+									]
+								},
+								"description": "This information allows FRAIHMWORK to track basic information about a system. Any extra or custom fields are expected to always be present even if null values are provided at first. Using this endpoint means that the system needs to be registered (under the hood), so a UUID will be created and assigned in the response"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:905]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 7]
```diff
@@ -0,0 +1,2038 @@
...
+									"raw": "{{baseUrl}}{{Mitigation}}/mitigation/",
+									"host": [
+										"{{baseUrl}}{{Mitigation}}"
+									],
+									"path": [
+										"mitigation",
+										""
+									]
+								},
+								"description": "This endpoint allows requesting services to find information about registered instances that FRAIHMWORK is currently aware of"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:961]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 8]
```diff
@@ -0,0 +1,2038 @@
...
+											],
+											"variable": [
+												{
+													"key": "uuid",
+													"value": "{{UUIDMitigationResponse}}",
+													"description": "(Required) Uuid of the mitigation posted"
+												}
+											]
+										},
+										"description": "This endpoint allows requesting services to find information about registered instances that FRAIHMWORK is currently aware of"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1025]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 9]
```diff
@@ -0,0 +1,2038 @@
...
+												":uuid"
+											],
+											"variable": [
+												{
+													"key": "uuid",
+													"value": "{{UUIDMitigationResponse}}"
+												}
+											]
+										},
+										"description": "This information allows FRAIHMWORK to track basic information about a system. Any extra or custom fields are expected to always be present even if null values are provided at first."
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1095]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 10]
```diff
@@ -0,0 +1,2038 @@
...
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1160]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 11]
```diff
@@ -0,0 +1,2038 @@
...
+											"host": [
+												"{{baseUrl}}{{Mitigation}}"
+											],
+											"path": [
+												"mitigation",
+												"state",
+												"RECOMMENDED"
+											]
+										},
+										"description": "This endpoint allows requesting services to find information about registered instances that FRAIHMWORK is currently aware of"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1220]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 12]
```diff
@@ -0,0 +1,2038 @@
...
+											"host": [
+												"{{baseUrl}}{{Mitigation}}"
+											],
+											"path": [
+												"mitigation",
+												"state",
+												"ACCEPTED"
+											]
+										},
+										"description": "This endpoint allows requesting services to find information about registered instances that FRAIHMWORK is currently aware of"
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1270]: Copy paste description. Should be corrected

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 13]
```diff
@@ -0,0 +1,2038 @@
...
+											"raw": "{{baseUrl}}{{Mitigation}}/subscription/mitigation",
+											"host": [
+												"{{baseUrl}}{{Mitigation}}"
+											],
+											"path": [
+												"subscription",
+												"mitigation"
+											]
+										},
+										"description": "Allows a system to report one or more faults currently affecting the system. Using this endpoint means that the client is not managing their own fault IDs, and the fault will require a new tracking UUID (that will be created and assigned in the response)."
```
Comment (GD-ResilienX) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1349]: Copy paste description. Should be corrected

Comment (GD-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1349]: This is the last one I'm commenting on. This whole file needs to be scrubbed for the documentation sections.

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 14]
```diff
@@ -0,0 +1,2038 @@
...
+							"item": [
+								{
+									"name": "GET By UUID",
+									"event": [
+										{
+											"listen": "test",
+											"script": {
+												"exec": [
+													"var jsonData = pm.response.json();"
+												],
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:194]: jsonData is not being used in this script. We can get rid of this.

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 15]
```diff
@@ -0,0 +1,2038 @@
...
+													"value": "{{access_token}}",
+													"type": "string"
+												}
+											]
+										},
+										"method": "GET",
+										"header": [],
+										"url": {
+											"raw": "{{baseUrl}}{{Monitoring}}/component/?:uuid={{UUIDComponentResponse}}",
+											"host": [
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:214]: get by uuid returns all the components instead of just one.

Comment (AS-resilienx) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:214]: Suggestion: {{baseUrl}}{{Monitoring}}/component/:uuid
Key: uuid
Value: {{UUIDComponentResponse}}

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 16]
```diff
@@ -0,0 +1,2038 @@
...
+									"response": []
+								}
+							]
+						},
+						{
+							"name": "PUT",
+							"item": [
+								{
+									"name": "Update mitigation to COMPLETED",
+									"event": [
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1036]: Changing mitigation state from RECOMMENDED to COMPLETE might not be possible. (Results in a bad request).
Update to ACCEPTED and then COMPLETE. 

Also, update the name of the test to "Update mitigation to COMPLETE".

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 17]
```diff
@@ -0,0 +1,2038 @@
...
+															]
+														}
+													},
+													"response": []
+												}
+											]
+										},
+										{
+											"name": "Cleanup",
+											"item": [
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1749]: the cleanup folder can be placed after the mitigation delete folder

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (MS-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1]: Good idea on the readme file. Added.

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json (hunk 17 above)
Comment (MS-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1749]: This cleanup is to demonstrate removing a subscription and moving the mitigation to a terminal state. It should not be at the same level of the Setup folder (at the top of the collection).

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json (hunk 16 above)
Comment (MS-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:1036]: I've reordered the two mitigation examples so ACCEPTED is prior to COMPLETE.

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 18]
```diff
@@ -0,0 +1,2049 @@
...
+									"response": []
+								}
+							]
+						},
+						{
+							"name": "PUT",
+							"item": [
+								{
+									"name": "Update existing properties about a Component Copy",
+									"event": [
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:243]: remove "Copy" from request name

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json [hunk 19]
```diff
@@ -0,0 +1,2049 @@
...
+									"response": []
+								}
+							]
+						},
+						{
+							"name": "/refresh",
+							"item": [
+								{
+									"name": "Refresh component Copy",
+									"event": [
```
Comment (AS-resilienx) [inline Integration/FRAIHMWORK Integrator Starter Kit Collection.json:376]: remove "Copy" from request name

**Comment by GD-ResilienX** [comment]
Recommend putting the REFRESH commands prior to the DELETE commands. These aren't necessarily meant to be run as a single collection in serial, but we should at least have it _work_ if it is

**Comment by GD-ResilienX** [comment]
Would also make sense (in the Faults and Mitigations sections) to have the instantiation of a sample component and fault as an environment setup folder to be used as needed. 

Both of these sets of requests depend on variables tracked from an earlier posted component or fault, so it is weird to expect them to be in isolation

**Review by GD-ResilienX — APPROVED** [review]
Made changes w.r.t my comments for Matt. Approve the rest!

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json (hunk 19 above)
Comment (GD-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:364]: Addressed! File scrubbed for all instances of word 'copy'

File: Integration/FRAIHMWORK Integrator Starter Kit Collection.json (hunk 18 above)
Comment (GD-ResilienX) [reply Integration/FRAIHMWORK Integrator Starter Kit Collection.json:243]: Addressed!

**Review by AS-resilienx — APPROVED** [review]

**Pull request merged**